docs/reinforced-concrete-hash.md

```diff
@@ -0,0 +1,342 @@
+  
+# Reinforced Concrete
+
+A fast and efficient [zk-friendly hash](../terms/zk-friendly-hash.md) function that leverages lookup gates to ensure security.
+
+References:
+
++ [Reinforced Concrete Paper](https://eprint.iacr.org/2021/1038.pdf)
+
+Other helpful resources:
+
++ [Reinforced Concrete Hash Slide](
+https://www.slideshare.net/AlexPruden/zkstudyclub-plonkup-reinforced-concrete-luke-pearson-joshua-fitzgerald-dmitry-khovratovich) .
++ [Reinforced Concrete and Plonkup Video](https://www.youtube.com/watch?v=Pnc9J7uQgqs&ab_channel=ZeroKnowledge).
++ [Reinforced Concrete by Dmitry Khovratovich](https://www.youtube.com/watch?v=SXnb7T9YATs&ab_channel=ZeroKnowledge).
+
+## Overview
+
+The Reinforced Concrete (RC) hash function works in the sponge framework, transforming a fixed-
+length bijective function into a variable-length hash.
+
+The RC permutation, shown in figure below, is a modified 7-round [SP network](
+https://en.wikipedia.org/wiki/Substitution%E2%80%93permutation_network), with elements from $\mathbb{F}_ p^3$ for a prime number $p$.
```

Comment:
Please explain what is SP network instead of giving only a reference link to Wikipedia